homeassistant/components/lupusec/alarm_control_panel.py

```diff
@@ -15,28 +16,23 @@
 )
 from homeassistant.core import HomeAssistant
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
-from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
 
 from . import DOMAIN as LUPUSEC_DOMAIN, LupusecDevice
 
 SCAN_INTERVAL = timedelta(seconds=2)
 
 
-def setup_platform(
+async def async_setup_entry(
     hass: HomeAssistant,
-    config: ConfigType,
-    add_entities: AddEntitiesCallback,
-    discovery_info: DiscoveryInfoType | None = None,
+    config_entry: ConfigEntry,
+    async_add_devices: AddEntitiesCallback,
 ) -> None:
     """Set up an alarm control panel for a Lupusec device."""
-    if discovery_info is None:
-        return
-
-    data = hass.data[LUPUSEC_DOMAIN]
+    data = hass.data[LUPUSEC_DOMAIN][config_entry.entry_id]
 
     alarm_devices = [LupusecAlarm(data, data.lupusec.get_alarm())]
```

Comment:
Is this async safe, ie no I/O or other blocking behavior?

Reply:
Is blocking. Will add fix to PR